### PR TITLE
Builds are only supported for GOOS=linux, so set it explicitly

### PR DESCRIPTION
**Pull request by mjlshen**
Signed-off-by: Michael Shen <mishen@umich.edu>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Since only target operating systems of linux are supported, so explicitly setting this will allow builds from other operating systems, like macOS to succeed. Furthermore, I noticed there was an error in the integrationtest make target.

**Which issue(s) this PR fixes**:
Related to #3322

```release-note
NONE
```

### CODE REVIEW

**Comment by gy95** [comment]
please sign off the commit, ref: https://github.com/kubeedge/kubeedge/pull/3325/checks?check_run_id=4268491656

**Comment by mjlshen** [comment]
> please sign off the commit, ref: https://github.com/kubeedge/kubeedge/pull/3325/checks?check_run_id=4268491656 😄

Oops, thanks, done! @gy95

File: hack/lib/golang.sh [hunk 1]
```diff
@@ -214,10 +214,9 @@ kubeedge::golang::build_binaries() {
     echo "building $bin"
     local name="${bin##*/}"
     set -x
-    go build -o ${KUBEEDGE_OUTPUT_BINPATH}/${name} -gcflags="${gogcflags:-}" -ldflags "${goldflags:-}" $bin
+    GOOS="linux" CGO_ENABLED=1 go build -o ${KUBEEDGE_OUTPUT_BINPATH}/${name} -gcflags="${gogcflags:-}" -ldflags "${goldflags:-}" $bin
```
Comment (gy95) [inline hack/lib/golang.sh:217]: for mac os, must this `CGO_ENABLED=1` be required?

File: Makefile [hunk 2]
```diff
@@ -143,7 +143,7 @@ integrationtest:
 	@echo "$$INTEGRATION_TEST_HELP_INFO"
 else
 integrationtest:
-	hack/make-rules/build.sh edgecore
+	hack/make-rules/build.sh edgecore; \
```
Comment (gy95) [inline Makefile:146]: could you please provide some materials about this `;`, is this required? :)

File: hack/lib/golang.sh (hunk 1 above)
Comment (mjlshen) [reply hack/lib/golang.sh:217]: Not needed for macOS, I saw that some of the binaries needed it. I can remove `CGO_ENABLED=1` if they don't all need CGO

File: Makefile (hunk 2 above)
Comment (mjlshen) [reply Makefile:146]: I think I was mistaken, I thought that `execute.sh` wasn't being run

Comment (fisherxu) [reply Makefile:146]: So what's meaning of the `; \` here?

File: hack/lib/golang.sh (hunk 1 above)
Comment (fisherxu) [reply hack/lib/golang.sh:217]: I think now we have no conclusion that kubeedge only supports Linux, so if we add `GOOS="linux"` here, all the other os   need to be tested.

File: Makefile (hunk 2 above)
Comment (mjlshen) [reply Makefile]: I reverted it, thought it was needed to run both commands

**Pull request closed without merging**